core-drawer-panel.html

```diff
@@ -75,344 +75,425 @@
+         * The panel to be selected when `core-drawer-panel` changes to narrow
+         * layout.
+         *
+         * @attribute defaultSelected
+         * @type string
+         * @default 'main'
+         */
+        defaultSelected: String,
+
+        /**
+         * Returns true if the panel is in narrow layout.  This is useful if you
+         * need to show/hide elements based on the layout.
+         *
+         * @attribute narrow
+         * @type boolean
+         * @default false
+         */
+        narrow: {
+          reflect: true,
+          type: String
+        },
+
+        /**
+         * If true, position the drawer to the right.
+         *
+         * @attribute rightDrawer
+         * @type boolean
+         * @default false
+         */
+        rightDrawer: Boolean,
+
+        /**
+         * If true, swipe to open/close the drawer is disabled.
+         *
+         * @attribute disableSwipe
+         * @type boolean
+         * @default false
+         */
+        disableSwipe: Boolean,
+
+        /**
+         * If true, ignore `responsiveWidth` setting and force the narrow layout.
+         *
+         * @attribute forceNarrow
+         * @type boolean
+         * @default false
+         */
+        forceNarrow: Boolean,
+
+        /**
+         * If true, swipe from the edge is disable.
+         *
+         * @attribute disableEdgeSwipe
+         * @type boolean
+         * @default false
+         */
+        disableEdgeSwipe: Boolean
+      },
+
+      listeners: {
+        click: 'onClick'
+      },
+
+      // TODO(ssorallen): Convert to 0.8 touch events
+      eventDelegates: {
+        trackstart: 'trackStart',
+        trackx: 'trackx',
+        trackend: 'trackEnd',
+        down: 'downHandler',
+        up: 'upHandler'
+      },
+
+      // Whether the transition is enabled.
+      transition: false,
+
+      // How many pixels on the side of the screen are sensitive to edge swipes and peek.
+      edgeSwipeSensitivity: 15,
+
+      // Whether the drawer is peeking out from the edge.
+      peeking: false,
+
+      // Whether the user is dragging the drawer interactively.
+      dragging: false,
+
+      // Whether the browser has support for the transform CSS property.
+      hasTransform: true,
+
+      // Whether the browser has support for the will-change CSS property.
+      hasWillChange: true,
+
+      // The attribute on elements that should toggle the drawer on tap, also
+      // elements will automatically be hidden in wide layout.
+      toggleAttribute: 'core-drawer-toggle',
+
+      configure: function() {
+        return {
+          defaultSelected: 'main',
+          disableEdgeSwipe: false,
+          drawerWidth: '256px',
+          forceNarrow: false,
+          narrow: false,
+          responsiveWidth: '640px',
+          rightDrawer: false,
+          selected: null
+        };
+      },
+
+      ready: function() {
+        this.hasTransform = 'transform' in this.style;
```

Comment:
This seems odd to have in this element; this is more the work of something like Modernizr. I can see many elements doing their own feature tests.